src/DTS/Repository/DocDataRepository.php

```diff
@@ -62,4 +64,17 @@ public function lookupId(string $docType, string $dataKey) {
     $row = $this->repoHelper->findNewest($filter, $cols);
     return $row['dataId'];
   }
-}
\ No newline at end of file
+
+  private function validate($type, $key) {
+    $errors = [];
+    if (empty($key)) {
+      $errors[] = "The key 'dataKey' is missing or empty";
```

Comment:
it isn't the job of a repository to talk to the client, that is for controllers and use cases

you could just throw a LengthException('dataKey may not be empty')

it is up to the user of this class to figure out what to tell the client if anything.